docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+===================================
+
+This view is a bit more involved than previous examples because we need to
+consider three different paths of execution:
+
+* get request
+* post request with invalid form (similar to get, render bound form with errors)
+* post request with a valid form (redirect to success url)
+
+The logic that we need to reuse is in the respective ``get_context_data()``
+methods and should all 'just work' as long as we set up the needed instance
+variables.
+
+The ``object_list`` needs to be set up in a custom way, by using
+``RelatedManager`` on ``object`` variable, therefore we need to initialize both
+instance variables before calling the ``get_context_data()`` methods.
+
+We could do that in ``get()`` and ``post()`` methods but if we look at
+execution paths, we can see that the first and second paths go through one
+common method: ``get_context_data()``, while third path does not need context
+data.
+
+All we have to do is override ``get_context_data()`` and set up the instance
+variables before making the ``super()`` call. This may seem a bit unusual
+because these variables are normally set up in ``get()`` or ``post()``, but in
+this case they have to be set up in both, and ``get_context_data()`` is the
+common execution path we can use to avoid duplicate logic.
+
+You may wish to access the object in ``form_valid()`` (e.g. for redirection);
+if you do, you have to use the ``get_object()`` method because
+``get_context_data()`` is not used in this execution path.
+
+(As a general note, it's good to remember the difference between
+``form_valid()`` and ``form_invalid()`` whenever you're working with FormView:
+though similarly named, their logic is completely different and it's useful to
+think of ``form_invalid()`` as 'retry get()' method).
+
+.. snippet::
+    :filename: views.py
+
+    class AuthorBooksView(SingleObjectMixin, MultipleObjectMixin, FormView):
+        model = Author
+        form_class = AuthorInterestForm
+        paginate_by = 2
+        template_name = "author-books.html"
+
+        def get_context_data(self, **kwargs):
+            self.object = self.get_object()
+            self.object_list = self.object.books.all()
+            return super(AuthorBooksView, self).get_context_data(**kwargs)
+
+        def form_valid(self, form):
+            # Here, we would have some logic based on user's form inputs
+            return redirect("author-books", pk=self.get_object().pk)
+
+
+.. code-block:: html+django
+
+    <h2>Author: {{ author.name }}</h2>
+
+    <ol>
+      {% for book in page_obj %}
+        <li>{{ book.name }}</li>
+      {% endfor %}
+    </ol>
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ form.as_table }}
+        </table>
+    <p><input type="submit" name="author_form" value="Submit" /></p>
+    </form>
+
+Combined Create and List Views
+==============================
```

Comment:
... finally this one I can't see an immediate problem with. The queryset is the same, using the pagination from the list view seems nice. I can actually see a case for including this in Django (with the necessary `get()` and `post()` implemented).